spring-boot-test-autoconfigure/src/test/resources/application.properties

```diff
@@ -0,0 +1,2 @@
+# Use random port for embedded Mongod.
+spring.data.mongodb.port = 0
```

Comment:
What about a BeanPostProcessor? Like

```
@Configuration
@EnableConfigurationProperties(MongoProperties.class)
@AutoConfigureBefore(EmbeddedMongoAutoConfiguration.class)
public class TestMongodAutoConfiguration {

	@Bean
	static MongoPropertiesBeanPostProcessor mongoPropertiesBeanPostProcessor() {
		return new MongoPropertiesBeanPostProcessor();
	}

	private static class MongoPropertiesBeanPostProcessor implements BeanPostProcessor {

		@Override
		public Object postProcessBeforeInitialization(Object bean, String beanName) throws BeansException {
			return bean;
		}

		@Override
		public Object postProcessAfterInitialization(Object bean, String beanName) throws BeansException {
			if (bean instanceof MongoProperties && ((MongoProperties) bean).getPort() == null) {
				((MongoProperties) bean).setPort(0);
			}
			return bean;
		}
	}
}
```

To slow? I'll checkout Gitter tomorrow… General Spring Boot room?